VoodooI2C/VoodooI2C/VoodooI2CDevice/VoodooI2CDeviceNub.cpp

```diff
@@ -88,9 +88,12 @@ IOReturn VoodooI2CDeviceNub::evaluateDSM(const char *uuid, UInt32 index, OSObjec
         OSArray::withCapacity(1),
     };
 
-    ret = acpi_device->evaluateObject("_DSM", result, params, 4);
-    if (ret != kIOReturnSuccess)
-        ret = acpi_device->evaluateObject("XDSM", result, params, 4);
+    ret = acpi_device->evaluateObject("XDSM", result, params, 4);
+    if (ret != kIOReturnSuccess) {
+        ret = acpi_device->evaluateObject("_DSM", result, params, 4);
```

Comment:
You should add a check for `ret` again after Line 93:
```
ret = acpi_device->evaluateObject("_DSM", result, params, 4);
if (ret != kIOReturnSuccess) {
....
}
```

Reply:
`evaluateDSM` is drafted as a wrapper for `evaluateObject`. The `IOReturn` result will be handled by methods calling it, which is `getDeviceResourcesDSM` in this case. So it's not necessary to verify it here.